site/_posts/2020-08-19-theme-registration-form.md

```diff
@@ -0,0 +1,285 @@
+
+### Updating the theme in FusionAuth
+
+After you've made all your changes to `defaultmessages.txt`, you need to turn it back into the JSON format the API expects. You must also add the keys and quotes previously stripped off. Finally, you have to make the update request to FusionAuth. For the update, use the `PATCH` method, which only updates the elements of an object provided. 
+
+This script performs all the above actions on the `defaultmessages.txt` file:
+
+```shell
+API_KEY=<your api key>
+THEME_ID=<your theme id>
+
+FILE_NAME=out.json$$
+
+awk '{printf "%s", $0"\\n"}' defaultmessages.txt |sed 's/^/{ "theme": { "defaultMessages": "/' | sed 's/$/"}}/' > $FILE_NAME
+
+STATUS_CODE=`curl -XPATCH -H 'Content-type: application/json' -H "Authorization: $API_KEY" 'http://localhost:9011/api/theme/'$THEME_ID -d @$FILE_NAME -o /dev/null -w '%{http_code}' -s`
+
+if [ $STATUS_CODE -ne 200 ]; then
+  echo "Error with patch, exited with status code: "$STATUS_CODE
+  exit 1
+fi
+
+rm $FILE_NAME
+```
+
+Run this script with the modified `defaultMessages.txt` file. This will apply the changes to your theme. 
+
+To view the changes, find the registration URL by going to the "Application" tab in the FusionAuth administrative user interface. View your application and copy the registration URL:
+
+{% include _image.liquid src="/assets/img/blogs/advanced-registration-forms/viewing-application-details.png" alt="Finding the registration URL." class="img-fluid" figure=false %}
+
+Open an incognito window and visit this URL. You should see your shiny new placeholders on the first page:
+
+{% include _image.liquid src="/assets/img/blogs/theming-advanced-forms/first-screen-themed.png" alt="The first page of the registration form with the correct messages added." class="img-fluid" figure=false %}
+
+If you fill out the first step, you'll see the second page with the correct hints as well:
+
+{% include _image.liquid src="/assets/img/blogs/theming-advanced-forms/second-screen-themed.png" alt="The second page of the registration form with the correct messages added." class="img-fluid" figure=false %}
+
+### Validation and theming
+
+On the second screen, if you submit the registration form without providing a price range, you're shown the error messages added above:
+
+{% include _image.liquid src="/assets/img/blogs/theming-advanced-forms/themed-with-validation-second-screen.png" alt="The second page of the registration form with error messages." class="img-fluid" figure=false %}
+
+## Adding form labels
+
+You can customize your field display more extensively by modifying the `customField` macro. This is in the "Helpers" section of your theme. Rather than use the API to do this, let's edit it directly in the administrative user interface.
+
+Navigate to "Themes" and edit your theme. Click on "Helpers" and scroll to the bottom. You'll be modifying the `customField` macro. Here's the default implementation for this freemarker macro:
```

Comment:
I think FreeMarker should be capitalized. Maybe even link to the website for it?